src/lib/toastr/toastr-service.ts

```diff
@@ -49,29 +49,31 @@ export class ToastrService {
     private _injector: Injector,
     private sanitizer: DomSanitizer,
   ) {
-    this.toastrConfig = this.applyConfig(toastrConfig);
+    const defaultConfig = new DefaultGlobalConfig;
+    this.toastrConfig = { ...defaultConfig, ...toastrConfig };
+    this.toastrConfig.iconClasses = { ...defaultConfig.iconClasses, ...toastrConfig.iconClasses };
   }
   /** show toast */
-  show(message?: string, title?: string, override?: IndividualConfig, type = '') {
+  show(message?: string, title?: string, override: Partial<IndividualConfig> = {}, type = '') {
```

Comment:
Is Partial on function parameters automatically optional?

Reply:
No, it's a required parameter unless made optional. I've made it optional by auto initializing it to an empty object if it isn't provided.

Reply:
oh right the default values duh